datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -1430,6 +1430,8 @@ impl PartialEq for Subquery {
     }
 }
 
+impl Eq for Subquery {}
```

Comment:
well I suppose `Eq` is satisfied given that PartialEq always returns false the requirements 

https://doc.rust-lang.org/std/cmp/trait.Eq.html

are satisfied in a strange sort of way